src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -252,6 +253,38 @@ public Object execute(
     return result;
   }
 
+  @Override
+  public Object executeOnConnection(Method method, List<Object> args)
```

Comment:
Does args need to be a list? `executeOnConnectionBoundObject` uses an array. If `executeOnConnection` takes in an array instead of list we can avoid the checks and conversion later in the code
```java
    Object[] argsArray = args == null ? null : args.toArray();
    performSpecialMethodHandlingIfRequired(argsArray, methodName);
```